### PR TITLE
[Generator] use is_sidepath:of:name if no name tagged

### PR DESCRIPTION
**Pull request by RedAuburn**
23k uses: 
https://taginfo.openstreetmap.org/keys/?key=is_sidepath:of:name#overview

### CODE REVIEW

File: generator/osm2type.cpp [hunk 1]
```diff
@@ -1478,6 +1478,10 @@ void GetNameAndType(OsmElement * p, FeatureBuilderParams & params,
       {
         params.SetDefaultNameIfEmpty(std::move(v));
       }},
+      {"is_sidepath:of:name", "*", [&params](string & k, string & v)
```
Comment (biodranik) [inline generator/osm2type.cpp:1481]: This block was designed for pistes according to the comment and implementation, right?
  // Fetch piste:name and piste:ref if there are no other name/ref values.

This code tries to set the name of the sidepath from the main street, right? Why?

Comment (RedAuburn) [reply generator/osm2type.cpp:1481]: > This block was designed for pistes according to the comment and implementation, right?

oops, moved to main block

> This code tries to set the name of the sidepath from the main street, right? Why?

No, it just sets the `name` to the value of the tag `is_sidepath:of:name` when `name` isn't already tagged. just a simple tag substitution like those in replaced_tags.txt :)

File: generator/osm2type.cpp [hunk 2]
```diff
@@ -1394,6 +1394,10 @@ void GetNameAndType(OsmElement * p, FeatureBuilderParams & params,
           params.layer = base::Clamp(params.layer, int8_t{feature::LAYER_LOW}, int8_t{feature::LAYER_HIGH});
         }
       }},
+      {"is_sidepath:of:name", "*", [&params](string & k, string & v)
```
Comment (biodranik) [inline generator/osm2type.cpp:1397]: Semantically, it should be: 
```suggestion
      {"is_sidepath:of:name", "*", [&params](std::string const & k, std::string && v)
```

@vng right?

File: generator/osm2type.cpp [hunk 3]
```diff
@@ -1394,6 +1394,10 @@ void GetNameAndType(OsmElement * p, FeatureBuilderParams & params,
...
+      {
+        params.SetDefaultNameIfEmpty(std::move(v));
```
Comment (biodranik) [inline generator/osm2type.cpp:1399]: Here the missing name of a sidewalk will be set from a nearby street, likely in the context of the recently discussed issue #8862

However, there could be issues with increased data size and with the search results. Also, only a few sidewalks have this tag set.

Would it be better to query nearby street names dynamically, only when needed?

Comment (vng) [reply generator/osm2type.cpp:1399]: Yes, it should be implemented in another way. This (PR) approach is very straightforward and can be confusing in other use-cases like map rendering or search.

Comment (RedAuburn) [reply generator/osm2type.cpp:1399]: agreed 👍️

**Pull request closed without merging**